api/src/routes/auth.ts

```diff
@@ -60,10 +63,25 @@ export const mobileAuth0Routes: FastifyPluginCallback = (
   // all auth routes.
   fastify.addHook('onRequest', fastify.redirectIfSignedIn);
 
-  fastify.get('/mobile-login', async req => {
+  fastify.get('/mobile-login', async (req, reply) => {
     const email = await getEmailFromAuth0(req);
 
-    await findOrCreateUser(fastify, email);
+    if (!email) {
+      return reply.status(401).send({
+        message: 'We could not log you in, please try again in a moment.',
+        type: 'danger'
+      });
+    }
```

Comment:
This is now being returned for `typeof email !== 'string'`, which is not happening on the old api.

Also, where does `401` come from? I agree it should probably be that, but the old api does `400`.

Reply:
Yeah, the idea was just that if auth0 responds with `{ ok: true, email: 1 }` or something equally weird, we have no way of knowing how to handle that.  Given that the mobile app only looks at the response cookies and not the body it's safe to respond with basically anything, so I figured a generic error response would be okay. @Nirajn2311 can you confirm/deny?

However, it probably makes more sense to throw for anything other than `{ ok: true, email: "some-string"}` and `{ ok: false}`, since those are the only two responses we can deal with.

As for the 401, it's a change, but the same argument applies: the mobile app does not care about status codes.